native/src/main/java/io/ballerina/xlibb/jsonpath/BJsonProvider.java

```diff
@@ -0,0 +1,272 @@
+        return JsonUtils.parse(jsonStream, charset);
+    }
+
+    /**
+     * Convert given json object to a json string
+     *
+     * @param obj object to transform
+     * @return json representation of object
+     */
+    @Override
+    public String toJson(Object obj) {
+        return StringUtils.getJsonString(obj);
+    }
+
+    /**
+     * Creates a provider specific json array
+     *
+     * @return new array
+     */
+    @Override
+    public Object createArray() {
+        return ValueCreator.createArrayValue(PredefinedTypes.TYPE_JSON_ARRAY);
+    }
+
+    /**
+     * Creates a provider specific json object
+     *
+     * @return new object
+     */
+    @Override
+    public Object createMap() {
+        return ValueCreator.createMapValue();
+    }
+
+    public boolean isArray(Object obj) {
+        return (obj instanceof BArray);
+    }
+
+    public Object getArrayIndex(Object obj, int idx) {
+        return toJsonArray(obj).get(idx);
+    }
+
+    public void setArrayIndex(Object array, int index, Object newValue) {
+        if (!isArray(array)) {
+            throw new UnsupportedOperationException();
+        } else {
+            BArray arrayNode = toJsonArray(array);
+            if (index == arrayNode.size()) {
+                arrayNode.append(createJsonElement(newValue));
+            } else {
+                arrayNode.add(index, createJsonElement(newValue));
+            }
+        }
+    }
+
+    public Object getMapValue(Object obj, String key) {
+        if (!(obj instanceof BMap)) {
+            throw new UnsupportedOperationException();
+        } else {
+            BMap jsonObject = toJsonObject(obj);
+            if (!jsonObject.containsKey(StringUtils.fromString(key))) {
+                return UNDEFINED;
+            } else {
+                return unwrap(jsonObject.get(StringUtils.fromString(key)));
+            }
+        }
+    }
+
+    public void setProperty(Object obj, Object key, Object value) {
+        if (isMap(obj)) {
+            toJsonObject(obj).put(key.toString(), createJsonElement(value));
+        } else {
+            BArray array = toJsonArray(obj);
+            long index;
+            if (key != null) {
+                index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+            } else {
+                index = array.size();
+            }
+
+            if (index == array.size()) {
+                array.append(createJsonElement(value));
+            } else {
+                array.add(index, createJsonElement(value));
+            }
+        }
+    }
+
+    public void removeProperty(Object obj, Object key) {
+        if (isMap(obj)) {
+            toJsonObject(obj).remove(key.toString());
+        } else {
+            BArray array = toJsonArray(obj);
+            int index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+            array.shift(index);
+        }
+    }
+
+    public boolean isMap(Object obj) {
```

Comment:
```suggestion
    private boolean isMap(Object obj) {
```

Reply:
This `isMap` is overided method. So we cannot make it private.